### ISSUE TITLE
[“Fantastic”] Smile Detector – The Quick Positive-Review Spotter

### ISSUE THREAD

**Issue by CodelineAtyab** [CodelineAtyab/OrbitXO#407]
# User Story: Sentiment-Analysis Tagging for Promotional Reviews

## Overview
As an **e-commerce Product Owner**,  
I want a **simple sentiment-analysis script** that automatically tags customer reviews as **“Positive”**,  
so that the team can *quickly surface happy feedback for promotional use*.

---

## Acceptance Criteria

| # | Criterion | Details |
|---|-----------|---------|
| **AC-1** | **Lightweight Training Dataset** | • ≥ 20 sample reviews<br>• Labeled **“Positive”** / **“Negative”**<br>• Delivered in **CSV** format |
| **AC-2** | **Single Python Script** | • May leverage **scikit-learn** (or similar)<br>• Trains a model on the dataset<br>• Exposes a **function** *or* simple **CLI**<br>• Returns **“Positive”** when a new review is predicted positive, **“Not Positive”** otherwise |
| **AC-3** | **Performance** | • End-to-end training on provided dataset completes in **< 60 s** on a standard laptop |
| **AC-4** | **Accuracy** | • Model achieves **≥ 75 %** accuracy on a **20 % hold-out** test split |
| **AC-5** | **Documentation** | • A concise **README** covering:<br>  – Setup<br>  – Execution<br>  – Adding more training data<br>• Total first-time setup ≤ 5 minutes |
| **AC-6** | **Licensing & Storage** | • All code & data reside in the shared repository<br>• Files include clear **MIT-style licensing headers** |

---

## Additional Notes

- **Dependencies**  
  • Python 3.8+  
  • scikit-learn, pandas, numpy (or equivalent lightweight libs)  

- **Output Format**  
  ```
  >>> predict_sentiment("Absolutely love this product—works like a charm!")
  "Positive"
  ```

- **Extensibility**  
  • Dataset can be expanded by appending new rows to the CSV.  
  • Model automatically retrains on the updated dataset when the script is re-run.

- **Definition of Done (DoD)**  
  • All Acceptance Criteria satisfied  
  • Continuous Integration job passes (linting, unit tests ≥ 90 % coverage)  
  • README verified by a new developer in ≤ 5 minutes  

---

© 2024 – Released under the MIT License.
